newsreader.py

```diff
@@ -25,65 +26,84 @@ def initConnection(self):
         for item in resp_categories:
             self.categories[item['id']] = item['name']
 
-        last = {}
+        # Read last stored post id
         try:
             with open(self.lfile) as f:
-                last = json.load(f)
+                self.last_post = int(f.read().strip())
         except Exception as e:
             print(e, datetime.datetime.now())
             traceback.print_exc()
-        self.groups = {}
-        res = self.conn.list()[1]
-        for g in res:
-            if g.group in last:
-                self.groups[g.group] = last[g.group]
-            else:
-                self.groups[g.group] = int(g.last)
+
+        # Update the last stored post id
+        resp = requests.get(self.conparams[0]+"posts.json", cookies=self.token)
```

Comment:
what if this request timeouts or doesn't result in 200(OK) response

this code block was previously writing down the upstream state if and only if a current state didn't exist (to create the `last file` if it didn't exist. but now it seems to be writing down upstream state no matter what. This might result in loss of messages, for example server starts, sees upstream has latest post id 20, updates internal state to 20, starts sending posts and crashes at 15th message, now when it restarts it will start from 20th message and skip 15 to 20.

could we change this code path to make use of upstream value iff `self.lfile` contains an invalid post id(or if file doesn't exist)